mvlearn/embed/cca.py

```diff
@@ -0,0 +1,261 @@
+
+        if self.signal_ranks is not None:
+            self.loadings_, scores, common_scores_normed, \
+                self.common_score_norms_, self.evals_ = _i_mcca(
+                    Xs,
+                    signal_ranks=self.signal_ranks,
+                    n_components=self.n_components,
+                    regs=self.regs,
+                    method=self.i_mcca_method,
+                )
+        else:
+            self.loadings_, scores, common_scores_normed, \
+                self.common_score_norms_, self.evals_ = _mcca_gevp(
+                    Xs,
+                    n_components=self.n_components,
+                    regs=self.regs
+                )
+        return scores, common_scores_normed
+
+    def stats(self, scores, stat=None):
+        r"""
+        Compute relevant statistics from the fitted CCA.
+
+        Parameters
+        ----------
+        scores: array-like, shape (2, n_samples, n_components)
+            The CCA scores.
+
+        stat : str, optional (default None)
+            The statistic to return. If None, returns a dictionary of all
+            statistics. Otherwise, specifies one of the following statistics
+
+            - 'r' : numpy.ndarray of shape (n_components,)
+                Canonical correlations of each component.
+
+            - 'Wilks' : numpy.ndarray of shape (n_components,)
+                Wilks' Lambda likelihood ratio statistic.
+
+            - 'df1' : numpy.ndarray of shape (n_components,)
+                Degrees of freedom for the chi-squared statistic, and
+                the numerator degrees of freedom for the F statistic.
+
+            - 'df2' : numpy.ndarray of shape (n_components,)
+                Denominator degrees of freedom for the F statistic.
+
+            - 'F' : numpy.ndarray of shape (n_components,)
+                Rao's approximate F statistic for H_0(k).
+
+            - 'pF' : numpy.ndarray of shape (n_components,)
+                Right-tail pvalue for stats['F'].
+
+            - 'chisq' : numpy.ndarray of shape (n_components,)
+                Bartlett's approximate chi-squared statistic for H_0(k)
+                with Lawley's modification.
+
+            - 'pChisq' : numpy.ndarray of shape (n_components,)
+                Right-tail pvalue for stats['chisq'].
+
+        Returns
+        -------
+        stats : dict or numpy.ndarray
+            Dict containing the statistics with keys specified above or
+            one of the statistics if specified by the `stat` parameter.
+        """
+        check_is_fitted(self)
+        scores = check_Xs(scores, enforce_views=2)
+        S1, S2 = scores
+        assert S1.shape[1] == S2.shape[1], \
+            "Scores from each view must have the same number of components."
+        n_components = S1.shape[1]
+
+        stats = {}
+
+        # pearson correlation coefficient
+        r = np.array([np.corrcoef(S1[:, i], S2[:, i])[0, 1]
```

Comment:
to avoid duplicating code, should the stats() function just use self.canon_corrs()